### DISCUSSION TITLE
Copy the files locally / Plex debrid like a BitTorrent client (for Radarr / Sonarr)

### DISCUSSION THREAD

**Discussion by raynoxfr** [itsToggle/plex_debrid#433] [Ideas]
Hello,

Excuse me for my English and mistakes...
So I have tested Plex Debrid, for few days, and it's great !
But I have some ideas (1 small and 1 big :D )

First of all, the small one...
I think if you let the user choose to copy the file to the local folder (for example from  webdav) and to connect/disconnect the webdav if not used (just for transporting the file)

The Second, it's the big one ....
Why don't you use your script to emulate like aria2 or BitTorrent client (and manage behind alldebrid, realdebrid etc)
You keep the strength of your solution, and you don't have to scrape, you let Overseerr (And behind Radarr / Sonarr) send the information to you and you "just" have to download (webdav, with alldebrid, etc) like BitTorrent client.
Of course, it's doesn't work with the Plex watchlist, this kind of implementation.

You have certainly, some explanation about your project ;)
Thank your for your answer

**Comment by berot3** [comment]
> Of course, it's doesn't work with the Plex watchlist, this kind of implementation.

That's what overseer does - keep track of your Plex watchlist.

**Comment by raynoxfr** [comment]
Yes, overseer can scrape the plex watchlist. But my approach is different in my explanations.
It's not the script that scrapes the watchlist, I let the tools do their job, as they already know how to do. The goal is to create a bridge between alldebrid and a bittorent client (already used by radarr / sonarr / etc)
That way the components keep running.
I am making a script, without Webdav, everything is done live (bridge).
I get the POST/GET request from radarr/sonarr/etc and then I parse the data and send it through the debrider API
It's a different approach, plex debrid have "given" me this idea
overseer -> sonarr / radarr / etc -> Script "simulate" bittorrent client (and communicate behind with debrid)
I have developed something (starting the previous week), still working, it's ready in the next days

**Reply by berot3** [reply]
have you checked out https://github.com/rogerfar/rdt-client ?

It seems like rdt is downloading everything from debrid. I actually dont want that. I want to stream it only. Not sure if thats possible with rdt-client.

**Reply by raynoxfr** [reply]
Thanks a lot, I have spent time on my script.... and it's exactly what I want, so I'm going to stop it... ;)

**Reply by berot3** [reply]
Haha no, never stop! You should do this! 🙂
But maybe you can contribute?

**Reply by itsToggle** [reply]
copying the files locally with a plex_debrid setup is as easy as moving them from the mounted folder to your harddrive. You can automate this too, but I dont see the point in this at all. Why not simply stream?
Building this only as a bittorrent client will remove some very crucial functionality, like the ability to sort scraped releases by their cache status. sonarr/radarr will pick releases independet of wether or not they are instantly available on your debrid service. There are a lot of other complications with this too, like watchlist integration and update intervals.
For me it was easier to simply write my own script which can do anything I want it to, instead of rewriting clumsy existing stuff like sonarr/radarr and a "fake" bittorrent client.

**Reply by berot3** [reply]
I get you point. I'm not too deep into it all. But there must be a reason people are using Rdt-client 😄

***

> You can automate this too

I guess with a script? Or is there already an existing solution?

> Why not simply stream?

Many people like to hord stuff and have full control. Keep it on their hard disk until there is no more internet 😂
But seriously, I am also into stream only, but sometimes I which I could organize my debrid-stuff better in Plex, I on a actual hard disk. 

> remove ... the ability to sort scraped releases by their cache status

Why? Can't the download client do this instead of sonarr?

> watchlist integration

Overseerr

> clumsy 

I like to use the adjective "essential" 😉

> "fake" bittorrent client

That's indeed a shame. A "own" solution would be probably better. Which would mean that radarr/sonarr would have to open up to a third general solution besides torrent and Usenet. That will most probably never happen. I think one reason is that debrid is seen as the enemy to torrent, since mostly doesn't seed back enough.

**Reply by itsToggle** [reply]
You can automate it by creating a backup or mirror routine in rclone, which is a built-in functionality.

You can organize your library with a plex_debrid setup. It's the exact same thing as a Google drive Plex server, except that storage it's a whole lot cheaper and there is no need wait for torrents to finish downloading.

No the download client can't do this, as sonarr and radarr are the ones doing the sorting.

There is no need for overseerr since Plex came out with Plex Discover.

Anyways, everything said here is already completely functional in plex_debrid.
